File: src/ControleFluxoEmpresarial/Models/Movimentos/ContaReceber.cs
using ControleFluxoEmpresarial.Entities;
using ControleFluxoEmpresarial.Models.CondicaoPagamentos;
using System;

namespace ControleFluxoEmpresarial.Models.Movimentos
{
    public class ContaReceber : IBaseAuditoria, IBaseEntity
    {
        public string Modelo { get; set; }
        public string Serie { get; set; }
        public string Numero { get; set; }
        public int Parcela { get; set; }
        public int ClienteId { get; set; }

        public decimal Valor { get; set; }
        public decimal? Desconto { get; set; }
        public decimal? Multa { get; set; }
        public decimal? Juro { get; set; }


        public string Descricao { get; set; }

        public int FormaPagamentoId { get; set; }
        public FormaPagamento FormaPagamento { get; set; }


        public DateTime DataVencimento { get; set; }
        public DateTime DataEmissao { get; set; }

        public DateTime? DataCancelamento { get; set; }
        public string UserCancelamento { get; set; }
        public string JustificativaCancelamento { get; set; }

        public DateTime? DataCancelamentoBaixa { get; set; }
        public string UserCancelamentoBaixa { get; set; }
        public string JustificativaCancelamentoBaixa { get; set; }

        public DateTime? DataBaixa { get; set; }
        public DateTime? DataPagamento { get; set; }
        public string UserBaixa { get; set; }
        public decimal? ValorBaixa { get; set; }

        public DateTime DataCriacao { get; set; }
        public string UserCriacao { get; set; }
        public DateTime? DataAtualizacao { get; set; }
        public string UserAtualizacao { get; set; }
    }
}

File: src/controle-fluxo-empresarial/src/models/Movimentos/ContaReceber.ts
import { IBaseEntity2 } from "../BaseEntity";
import { FormaPagamento } from '../CondicaoPagamento/FormaPagamento';
import { Cliente } from './../Pessoas/Cliente';

export default interface ContaReceber extends IBaseEntity2 {
    numero?: string | null;
    modelo?: string | null;
    serie?: string | null;
    
    valor?: number | null;
    desconto?: number | null;
    multa?: number | null;
    juro?: number | null;
    parcela?: number | null;
    
    descricao?: string | null;
    
    dataBaixa?: Date | null;
    dataEmissao?: Date | null;
    dataPagamento?: Date | null;
    dataCancelamento?: Date | null;
    dataVencimento?: Date | null;

    clienteId?: number | null;
    cliente?: Cliente | null;

    formaPagamentoId?: number | null;
    formaPagamento?: FormaPagamento | null;
}

export interface CancelarContaReceber {
    numero?: string | null;
    modelo?: string | null;
    serie?: string | null;
    parcela?: number | null;

    justificativa?: string | null;
    senha?: string | null;
}